src/auth.js

```diff
@@ -57,20 +57,20 @@ const generate_password = function (length) {
 	}
 };
 
-const signUp = ({ phone_number, display_name }) => {
+const signUp = (phone_number, display_name) => {
 	const client = new CognitoIdentityProviderClient(cognito);
 
 	const command = new SignUpCommand({
 		ClientId: cognito.userPoolClientId,
 		Username: phone_number,
 		Password: generate_password(30),
-		UserAttributes: [{ Name: "custom:display_name", Value: display_name }],
+		UserAttributes: [{ Name: "preferred_username", Value: display_name }],
```

Comment:
since preferred_username is built in, it might just be PreferredUsername and not UserAttributes...

Reply:
Yeah, the [documentation](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/cognito-identity-provider/command/SignUpCommand/) isn't 100% clear, but I think it's under UserAttributes.

Reply:
https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html

this is what I was thinking of